GitHubTeams.ps1

```diff
@@ -276,6 +314,407 @@ filter Get-GitHubTeamMember
     return (Invoke-GHRestMethodMultipleResult @params | Add-GitHubUserAdditionalProperties)
 }
 
+function New-GitHubTeam
+{
+<#
+    .SYNOPSIS
+        Creates a team within an organization on GitHub.
+
+    .DESCRIPTION
+        Creates a team or within an organization on GitHub.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OrganizationName
+        The name of the organization to create the team in.
+
+    .PARAMETER TeamName
+        The name of the team.
+
+    .PARAMETER Description
+        The description for the team.
+
+    .PARAMETER Maintainers
+        A list of GitHub IDs for organization members who will become team maintainers.
+
+    .PARAMETER RepositoryName
+        A list of repositories to add the team to.
+
+    .PARAMETER Privacy
+        The level of privacy this team should have.
+
+    .PARAMETER ParentTeamName
```

Comment:
I think it would make sense for the parent team to be targetable by the pipeline once implemented.  So, specifying either the ParentTeamName (not via pipeline) or the ParentTeamSlug (via `PipelineByPropertyName`)

Reply:
Your right there.  At the time that I wrote that I must have forgotten that `TeamName` / 'TeamSlug' was already a parameter.

Reply:
Ah, just re-looked at the code and realized what I was saying and I stand by it.

This should have:
   * `ParentTeamName` - Just a normal parameter with no pipeline support
   * `ParentTeamSlug` (aliased to `TeamSlug`) - A parameter with `ValueFromPipelineByPropertyname`.  This way, someone can pipe in a parent team and have it be the parent and avoid the extra call needed to resolve the parent team by name.

Reply:
Or actually, looking at the code again more speficially, `ParentTeamId` (aliased to `[TeamId`]).  But, logic otherwise follows suit.

Reply:
But then you would get the option of 'TeamSlug' for parameter completion of the function, which would actually be the ParentTeam. Pretty ugly.

Reply:
This would need adding in another PR.